actionmailer/lib/action_mailer/base.rb

```diff
@@ -944,6 +831,115 @@ def self.supports_path?
       false
     end
 
+  private
+
+    def default_values
+      @_default_values ||= self.class.default.map do |key, value|
+        [
+          key,
+          value.is_a?(Proc) ? instance_eval(&value) : value
+        ]
+      end.to_h
+    end
+
+    def content_type_for_message_with_attachments
+      [
+        "multipart",
+        message.attachments.detect(&:inline?) ? "related" : "mixed",
+        message.content_type_parameters || {}
+      ]
+    end
+
+    def content_type_for_multipart_message
+      [
+        "multipart",
+        "alternative",
+        message.content_type_parameters || {}
+      ]
+    end
+
+    def handle_parts_order
```

Comment:
This method has an implitcit dependency on `process_headers_from_user` being called first to set `@headers`. Instead of that we should receive `headers` as argument.